src/lib/main.js

```diff
@@ -81,33 +74,48 @@ function getAllHolidays(url) {
  *
  * @returns Promise
  */
-function authenticate(url) {
-  return new Promise((resolve, reject) => {
-    tabs.open({
-      url,
-      inBackground: true,
-      onReady: function(tab) {
-        if (tab.title.startsWith('401')) {
-          reject(new Error('User canceled authentication'));
-        } else {
-          resolve();
-        }
-        tab.close();
+async function authenticate(url) {
+  const tab = await tabs.create({ url });
+
+  // Note: tab.status is invalid right now, we need to wait for a first
+  // onUpdated callback to get it right.
+  // See https://bugzilla.mozilla.org/show_bug.cgi?id=1381756
+
+  await new Promise((resolve, reject) => {
+    function removedListener(id) {
+      if (id !== tab.id) {
+        return;
       }
-    });
+      tabs.onRemoved.removeListener(removedListener);
+      tabs.onUpdated.removeListener(updatedListener);
+      reject(new Error('User closed the authentication tab.'));
+    }
+
+    function updatedListener(id, _, updatedTab) {
+      if (id !== tab.id) {
+        return;
+      }
+
+      if (updatedTab.url === url) {
+        tabs.onRemoved.removeListener(removedListener);
+        tabs.onUpdated.removeListener(updatedListener);
+        resolve();
+      }
+    }
+
+    tabs.onRemoved.addListener(removedListener);
+    tabs.onUpdated.addListener(updatedListener);
   });
+
+  await tabs.remove(tab.id);
```

Comment:
Can we reuse the same tab to open the results page without closing it? Maybe by adding a Loading page in between so that we can keep using fetch for the data parsing?